sunpy/physics/limb_darkening.py

```diff
@@ -117,30 +108,29 @@ def get_limb_dark(sun_map):
     ---------
     Cox, Arthur N., Allen's astrophysical quantities, 4th ed. Publisher:
         New York: AIP Press; Springer, 2000. ISBN: 038798746
-
+    DOI : https://doi.org/10.1063/1.1325201
     """
 
     if not (4000 * u.AA < sun_map.wavelength < 15000 * u.AA):
-        raise ValueError("The wavelength of the observation \
-                          must be between 4000 and 15000 Angstrom.")
+        raise ValueError("The wavelength of the observation " +
+                          "must be between 4000 and 15000 Angstrom.")
 
-    ul = darklimb_u(sun_map.wavelength)
-    vl = darklimb_v(sun_map.wavelength)
+    ul = __darklimb_u(sun_map.wavelength)
+    vl = __darklimb_v(sun_map.wavelength)
 
     x_center, y_center = sun_map.world_to_pixel(SkyCoord(0, 0, unit=u.arcsec, frame=sun_map.coordinate_frame))
-    x_center = x_center.to(u.pixel)
-    y_center = y_center.to(u.pixel)
-    x_dim = sun_map.dimensions.x.to(u.pixel).value
-    y_dim = sun_map.dimensions.y.to(u.pixel).value
-    radius = (sun_map.rsun_obs.to(u.arcsec).value /
-              sun_map.scale.axis1.to(u.arcsec / u.pixel).value)
+    x_center = x_center
+    y_center = y_center
+    x_dim = sun_map.dimensions.x
+    y_dim = sun_map.dimensions.y
+    radius = sun_map.rsun_obs / sun_map.scale.axis1
 
-    x_grid, y_grid = np.meshgrid(np.arange(int(x_dim)), np.arange(int(y_dim)), sparse=True)
+    x_grid, y_grid = np.meshgrid(np.arange(x_dim.value), np.arange(y_dim.value), sparse=True)
```

Comment:
You should always use `to_value` to be explicit about what units you are dropping.